uber/config.py

```diff
@@ -412,3 +412,9 @@ def _is_intstr(s):
     c.VOLUNTEER_CHECKLIST = [url for step, url in _items]
 
 stripe.api_key = c.STRIPE_SECRET_KEY
+
+if c.KIOSK_CC_ENABLED:
+    c.DOOR_PAYMENT_METHOD_OPTS.append([c.STRIPE, "Pay with credit card now (faster)"])
+
+if c.GROUPS_ENABLED:
+    c.DOOR_PAYMENT_METHOD_OPTS.append([c.GROUP, "Taking an unassigned Group badge (group leader must be present)"])
```

Comment:
Is c.STRIPE existing in this example with some hashed integer value intended, or a side effect?

> On Oct 7, 2015, at 3:45 PM, Eli Courtwright <notifications@github.com> wrote:
> 
> In uber/config.py:
> 
> > @@ -412,3 +412,9 @@ def _is_intstr(s):
> >      c.VOLUNTEER_CHECKLIST = [url for step, url in _items]
> >  
> >  stripe.api_key = c.STRIPE_SECRET_KEY
> > +
> > +if c.KIOSK_CC_ENABLED:
> > +    c.DOOR_PAYMENT_METHOD_OPTS.append([c.STRIPE, "Pay with credit card now (faster)"])
> > +
> > +if c.GROUPS_ENABLED:
> > +    c.DOOR_PAYMENT_METHOD_OPTS.append([c.GROUP, "Taking an unassigned Group badge (group leader must be present)"])
> The code that you've written here references the c.STRIPE constant and the c.GROUP constant. It uses them to add key/value pairs to the c.DOOR_PAYMENT_METHOD_OPTS list. However, if you entered these if-blocks and c.STRIPE or c.GROUP were not already defined, you'd get an exception.
> 
> Many of the settings on c are defined by virtue of being in an enum subsection in our config. In other words, if I have
> 
> [enums]
> [[thing]]
> foo = 'Fu'
> bar = 'Bahr'
> then not only will we have c.THINGS, c.THING_OPTS and c.THING_VARS, we'll also have a c.FOO and a c.BAR constant defined. Sometimes the same option name will show up in multiple different enums, e.g. we have a c.ARCADE which is referred to in several different enum subsections, and it will always have the same value (since the integer constant is derived from a hash of the constant name).
> 
> My point here is that you've removed the group = ... and stripe = ... from the [[door_payment_method]] enum subsection, which means that the only way that c.GROUP and c.STRIPE will even exist is if they also happen to be defined elsewhere. Which I'm guessing they are, or else you'd have seen an exception when testing this out.
> 
> I just wanted to make sure we were on the same page about that, since I hadn't had a chance to actually read through configspec.ini to confirm that we're guaranteed to always have these constants defined.
> 
> —
> Reply to this email directly or view it on GitHub.
>